### PR TITLE
fix: require passing repo options to migrator

### PR DESCRIPTION
**Pull request by achingbrain**
Adds the changes released in `1.0.1` to master.

BREAKING CHANGES:

- `migrator.migrate(path, version, opts)` has changed to `migrator.migrate(path, repoOpts, version, opts)`
- `migrator.revert(path, version, opts)` has changed to `migrator.revert(path, repoOpts, version, opts)`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,11 +20,9 @@
   "main": "src/index.js",
   "browser": {
     "./src/repo/lock.js": "./src/repo/lock-memory.js",
+    "./src/repo/default-root-options.js": "./src/repo/ldefault-root-options.browser.js",
```
Comment (jacobheun) [inline package.json]: Not seeing these files anywhere

File: README.md [hunk 2]
```diff
@@ -127,35 +130,24 @@ Signature of the progress callback.
  * `counter` (int) - index of current migration.
  * `totalMigrations` (int) - total count of migrations that will be run.
 
-### `.revert(path, toVersion, {ignoreLock, repoOptions, onProgress, isDryRun}) -> Promise<void>`
+### `.revert(path, repoOptions, toVersion, {ignoreLock, onProgress, isDryRun}) -> Promise<void>`
```
Comment (jacobheun) [inline README.md:133]: `.migrate` also needs to be updated to reflect this.

File: test/index.spec.js [hunk 3]
```diff
@@ -44,6 +44,12 @@ function createOptions () {
   }
 }
 
+function createRepoOptions () {
+  return {
+
+  }
+}
```
Comment (jacobheun) [inline test/index.spec.js:51]: Why is this empty? Options are required but can be an empty object? That seems silly.

File: package.json (hunk 1 above)
Comment (achingbrain) [reply package.json]: Fixed

File: README.md (hunk 2 above)
Comment (achingbrain) [reply README.md:133]: Fixed

File: test/index.spec.js (hunk 3 above)
Comment (achingbrain) [reply test/index.spec.js:51]: The implementation is a sinon stub so it doesn't do any verification, we just validate that we were passed *something* in the test.

**Review by jacobheun — APPROVED** [review]
Pending ci this lgtm

**Pull request merged**